src/profile-logic/marker-data.js

```diff
@@ -235,103 +233,6 @@ export function getTabFilteredMarkerIndexes(
-  const newMarkers: RawMarkerTable = {
-    data: markers.data.slice(),
-    name: markers.name.slice(),
-    startTime: markers.startTime.slice(),
-    endTime: markers.endTime.slice(),
-    phase: markers.phase.slice(),
-    category: markers.category.slice(),
-    length: markers.length,
-  };
-
-  // Match: "Bailout_MonitorTypes after add on line 1013 of self-hosted:1008"
-  // Match: "Bailout_TypeBarrierO at jumptarget on line 1490 of resource://devtools/shared/base-loader.js -> resource://devtools/client/shared/vendor/immutable.js:1484"
-  // Match: "Bailout_ShapeGuard at jumptarget on line 7 of moz-extension://<URL>"
-  const bailoutRegex =
-    // Capture groups:
-    //       type   afterAt    where        bailoutLine  script     functionLine
-    //        ↓     ↓          ↓                  ↓        ↓        ↓
-    /^Bailout_(\w+) (after|at) ([\w _-]+) on line (\d+) of (.*?)(?::(\d+))?$/;
-  //                                                               ↑
-  // Colon in non-capturing group which only matches when functionLine is present
-
-  // Match: "Invalidate resource://devtools/shared/base-loader.js -> resource://devtools/client/shared/vendor/immutable.js:3662"
-  // Match: "Invalidate self-hosted:4032"
-  // Match: "Invalidate moz-extension://<URL>"
-  const invalidateRegex =
-    // Capture groups:
-    //         url        line
-    //           ↓        ↓
-    /^Invalidate (.*?)(?::(\d+))?$/;
-
-  const bailoutStringIndex = stringTable.indexForString('Bailout');
-  const invalidationStringIndex = stringTable.indexForString('Invalidate');
-  for (let markerIndex = 0; markerIndex < markers.length; markerIndex++) {
-    const nameIndex = markers.name[markerIndex];
-    const name = stringTable.getString(nameIndex);
-    let matchFound = false;
-    if (name.startsWith('Bailout_')) {
-      matchFound = true;
-      const match = name.match(bailoutRegex);
-      if (!match) {
-        console.error(`Could not match regex for bailout: "${name}"`);
-      } else {
-        const [
-          ,
-          type,
-          afterAt,
-          where,
-          bailoutLine,
-          script,
-          functionLine,
-        ] = match;
-        newMarkers.name[markerIndex] = bailoutStringIndex;
-        newMarkers.data[markerIndex] = ({
-          type: 'Bailout',
-          bailoutType: type,
-          where: afterAt + ' ' + where,
-          script: script,
-          bailoutLine: +bailoutLine,
-          functionLine: functionLine === undefined ? null : +functionLine,
-        }: BailoutPayload);
-      }
-    } else if (name.startsWith('Invalidate ')) {
-      matchFound = true;
-      const match = name.match(invalidateRegex);
-      if (!match) {
-        console.error(`Could not match regex for invalidation: "${name}"`);
-      } else {
-        const [, url, line] = match;
-        newMarkers.name[markerIndex] = invalidationStringIndex;
-        newMarkers.data[markerIndex] = ({
-          type: 'Invalidation',
```

Comment:
Wow, this is super confusing that we were _renaming_ this marker from `Invalidate` to `Invalidation`. Another weirdness gone then...

Reply:
More weird handling gone!